graylog2-web-interface/src/views/components/MigrateFieldCharts.jsx

```diff
@@ -0,0 +1,207 @@
+import { maxBy, values, isEmpty } from 'lodash';
+import Immutable from 'immutable';
+
+import { CurrentViewStateStore } from 'views/stores/CurrentViewStateStore';
+import { ViewStatesActions } from 'views/stores/ViewStatesStore';
+import SearchActions from 'views/actions/SearchActions';
+
+import Store from 'logic/local-storage/Store';
+import { widgetDefinition } from 'views/logic/Widgets';
+import AggregationWidget from 'views/logic/aggregationbuilder/AggregationWidget';
+import AggregationWidgetConfig from 'views/logic/aggregationbuilder/AggregationWidgetConfig';
+import Pivot from 'views/logic/aggregationbuilder/Pivot';
+import Series from 'views/logic/aggregationbuilder/Series';
+import WidgetPosition from 'views/logic/widgets/WidgetPosition';
+import LineVisualizationConfig from 'views/logic/aggregationbuilder/visualizations/LineVisualizationConfig';
+import AreaVisualizationConfig from 'views/logic/aggregationbuilder/visualizations/AreaVisualizationConfig';
+import type { InterpolationMode } from 'views/logic/aggregationbuilder/visualizations/Interpolation';
+
+import { Alert, Button, Row, Col } from 'components/graylog';
+import Spinner from 'components/common/Spinner';
+
+type LegacySeries = 'mean' | 'max' | 'min' | 'total' | 'count' | 'cardinality';
+type LegacyInterpolation = 'linear' | 'step-after' | 'basis' | 'bundle' | 'cardinal' | 'monotone';
+type LegacyInterval = 'minute' | 'hour' | 'day' | 'week' | 'month' | 'quarter' | 'year'
+type LegacyVisualization = 'bar' | 'area' | 'line' | 'scatterplot'
+type LegacyFieldChart = {
+  field: string,
+  renderer: LegacyVisualization,
+  interpolation: LegacyInterpolation,
+  valuetype: LegacySeries,
+  interval: LegacyInterval,
+}
+
+const Actions = styled.div`
+  margin-top: 10px;
+`;
+
+const mapSeries = (legacySeries: LegacySeries, field: string) => {
+  let seriesName;
+  switch (legacySeries) {
+    case 'total':
+      seriesName = 'sum';
+      break;
+    case 'mean':
+      seriesName = 'avg';
+      break;
+    case 'cardinality':
+      seriesName = 'card';
+      break;
+    default:
+      seriesName = legacySeries;
+  }
+  return `${seriesName}(${field})`;
+};
+
+const mapVisualization = (legacyVisualization: LegacyVisualization) => {
+  switch (legacyVisualization) {
+    case 'scatterplot':
+      return 'scatter';
+    default:
+      return legacyVisualization;
+  }
+};
+
+const mapTime = (legacyTime: string) => {
+  switch (legacyTime) {
+    case 'quarter':
+      return { unit: 'months', value: 3 };
+    default:
+      return { unit: `${legacyTime}s`, value: 1 };
+  }
+};
+
+const createVisualizationConfig = (legacyInterpolation: LegacyInterpolation, visualization: string) => {
+  let interpolation: InterpolationMode;
+  switch (legacyInterpolation) {
+    case 'basis':
+    case 'bundle':
+    case 'cardinal':
+    case 'monotone':
+      interpolation = 'spline';
+      break;
+    case 'linear':
+    case 'step-after':
+      interpolation = legacyInterpolation;
```

Comment:
After extracting `mapInterpolation` you could just return `legacyInterpolation` for both cases here.